### PR TITLE
feat: added download all documents as a zip folder

### PR DESCRIPTION
**Pull request by isaackps**
# in this PR:

- Added a button to download all successfully issued documents

Screenshot 2020-10-06 at 4 59 36 PM

### CODE REVIEW

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.test.tsx [hunk 1]
```diff
@@ -156,4 +160,25 @@ describe("publishedScreen", () => {
     expect(screen.queryAllByText(/Please wait while we prepare your document/)).toHaveLength(1);
     expect(screen.queryAllByTestId("publish-loader")).toHaveLength(0);
   });
+
+  it("should download the file correctly when generateZipFile method is called", async () => {
+    render(
+      <PublishedScreen
+        publishedDocuments={mockPublishedDocuments}
+        failedPublishedDocuments={mockFailPublishedDocuments}
```
Comment (simboonlong) [inline src/components/PublishContainer/PublishedScreen/PublishedScreen.test.tsx:168]: can pass in empty array?

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx [hunk 2]
```diff
@@ -42,6 +44,19 @@ export const PublishedScreen: FunctionComponent<PublishScreen> = ({
     };
   });
 
+  const generateZipFile = (): void => {
+    const zip = new JSZip();
+    publishedDocuments.forEach((document) => {
+      const file = JSON.stringify(document.wrappedDocument);
```
Comment (simboonlong) [inline src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx:50]: if u want pretty json:

```suggestion
      const file = JSON.stringify(document.wrappedDocument, null, 2);
```

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx [hunk 3]
```diff
@@ -42,6 +44,19 @@ export const PublishedScreen: FunctionComponent<PublishScreen> = ({
...
+      const blob = new Blob([file], { type: "text/json;charset=utf-8" });
+      zip.file(generateFileName(config, document.fileName, "tt"), blob);
+    });
+
+    zip.generateAsync({ type: "blob" }).then((content) => {
+      saveAs(content, generateFileName(config, "TradeTrust-Documents", "zip"));
```
Comment (simboonlong) [inline src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx:56]: i think can name as just `Documents` will do? cos creator is quite universal, can be use for other stuff i think

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.test.tsx (hunk 1 above)
Comment (isaackps) [reply src/components/PublishContainer/PublishedScreen/PublishedScreen.test.tsx:168]: ok can

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx (hunk 2 above)
Comment (isaackps) [reply src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx:50]: ok

File: src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx (hunk 3 above)
Comment (isaackps) [reply src/components/PublishContainer/PublishedScreen/PublishedScreen.tsx:56]: ok sure, will update it

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**